lib/rubocop/formatter/disabled_config_formatter.rb

```diff
@@ -137,13 +137,14 @@ def output_cop_comments(output_buffer, cfg, cop_name, offense_count)
         output_cop_param_comments(output_buffer, params, default_cfg)
       end
 
-      def supports_safe_auto_correct?(cop_class, default_cfg)
+      def supports_safe_autocorrect?(cop_class, default_cfg)
         cop_class&.support_autocorrect? &&
-          (default_cfg.nil? || default_cfg['Safe'] || default_cfg['Safe'].nil?)
+          (default_cfg.nil? || (safe?(default_cfg) && safe_autocorrect?(default_cfg)))
```

Comment:
Combined with the other comments, I think the condition can be written like this.

```diff
       def supports_safe_autocorrect?(cop_class, default_cfg)
-        cop_class&.support_autocorrect? &&
-          (default_cfg.nil? || (safe?(default_cfg) && safe_autocorrect?(default_cfg)))
+        cop_class&.support_autocorrect? && safe_autocorrect?(default_cfg)
       end
```